config/main.py

```diff
@@ -3153,111 +3214,110 @@ def del_vrf_vni_map(ctx, vrfname):
 @click.pass_context
 def add_route(ctx, command_str):
     """Add route command"""
-    if len(command_str) < 4 or len(command_str) > 9:
-        ctx.fail("argument is not in pattern prefix [vrf <vrf_name>] <A.B.C.D/M> nexthop <[vrf <vrf_name>] <A.B.C.D>>|<dev <dev_name>>!")
-    if "prefix" not in command_str:
-        ctx.fail("argument is incomplete, prefix not found!")
-    if "nexthop" not in command_str:
-        ctx.fail("argument is incomplete, nexthop not found!")
-    for i in range(0, len(command_str)):
-        if "nexthop" == command_str[i]:
-            prefix_str = command_str[:i]
-            nexthop_str = command_str[i:]
-    vrf_name = ""
-    cmd = 'sudo vtysh -c "configure terminal" -c "ip route'
-    if prefix_str:
-        if len(prefix_str) == 2:
-            prefix_mask = prefix_str[1]
-            cmd += ' {}'.format(prefix_mask)
-        elif len(prefix_str) == 4:
-            vrf_name = prefix_str[2]
-            prefix_mask = prefix_str[3]
-            cmd += ' {}'.format(prefix_mask)
-        else:
-            ctx.fail("prefix is not in pattern!")
-    if nexthop_str:
-        if len(nexthop_str) == 2:
-            ip = nexthop_str[1]
-            if vrf_name == "":
-                cmd += ' {}'.format(ip)
-            else:
-                cmd += ' {} vrf {}'.format(ip, vrf_name)
-        elif len(nexthop_str) == 3:
-            dev_name = nexthop_str[2]
-            if vrf_name == "":
-                cmd += ' {}'.format(dev_name)
-            else:
-                cmd += ' {} vrf {}'.format(dev_name, vrf_name)
-        elif len(nexthop_str) == 4:
-            vrf_name_dst = nexthop_str[2]
-            ip = nexthop_str[3]
-            if vrf_name == "":
-                cmd += ' {} nexthop-vrf {}'.format(ip, vrf_name_dst)
+    config_db = ctx.obj['config_db']
+    key, route = cli_sroute_to_config(ctx, command_str)
+
+    # If defined intf name, check if it exists
+    if 'ifname' in route:
+        if (not route['ifname'] in config_db.get_keys('VLAN') and
+            not route['ifname'] in config_db.get_keys('INTERFACE') and
+            not route['ifname'] in config_db.get_keys('PORTCHANNEL')):
+            ctx.fail('interface {} doesn`t exist'.format(route['ifname']))
+
+    # Check if exist entry with key
+    keys = config_db.get_keys('STATIC_ROUTE')
+    if key in keys:
+        # If exist update current entry
+        current_entry = config_db.get_entry('STATIC_ROUTE', key)
+
+        for entry in ['nexthop', 'nexthop-vrf', 'ifname']:
+            if not entry in current_entry:
+                    current_entry[entry] = ''
+            if entry in route:
+                current_entry[entry] += ',' + route[entry]
             else:
-                cmd += ' {} vrf {} nexthop-vrf {}'.format(ip, vrf_name, vrf_name_dst)
-        else:
-            ctx.fail("nexthop is not in pattern!")
-    cmd += '"'
-    clicommon.run_command(cmd)
+                current_entry[entry] += ','
```

Comment:
Just wanted to confirm, does this command only allow to add one nexthop at a time and something like `config route add prefix 6.2.3.4/32 nexthop "30.0.0.6,30.0.0.7"` is not valid?

Reply:
This is covered in the example and should be supported

Reply:
Then this is a problem. If a command is issued like this `config route add prefix 6.2.3.4/32 nexthop "30.0.0.6,30.0.0.7"`, the nexthop field would add `,30.0.0.6,30.0.0.7` (two additional values), whereas `ifname` and `nexthop-vrf` would only add `,` (one additional value). It follows that the number of values in the fields would mismatch after calling the command.

Reply:
For this command, I wouldn't expect any `ifname `or `nexthop-vrf` attribute since its not mentioned in the CLI. Can you please check `test_multiple_nexthops_static_route(self):.` It looks good to me.

Reply:
`test_multiple_nexthops_static_route` only covers the scenario of adding a new entry, but this part of the code is to add to existing entries. I still doubt if this would work. For example, if we call the command in the following sequence: `config route add prefix 6.2.3.4/32 nexthop "30.0.0.5"` and then ``config route add prefix 6.2.3.4/32 nexthop "30.0.0.6,30.0.0.7"``. The corresponding entry in CONFIG_DB would be `"STATIC_ROUTE":{"6.2.3.4/32": {"nexthop": "30.0.0.5,30.0.0.6,30.0.0.7", "ifname": ",", "nexthop-vrf": ","}}`. This is invalid since `nexthop` has three values and others only have two. I would suggest to fix this and cover the scenario in test.

Reply:
Agree. I thought you were referring to first time route add based on previous comment. @d-dashkov to address this.

Reply:
Fixed. Please check test_multiple_nexthops_static_route.
I added alignment for such a case and now ```config route add prefix 6.2.3.4/32 nexthop "30.0.0.6,30.0.0.7"``` creates
 ```{'nexthop': '30.0.0.6,30.0.0.7', 'blackhole': 'false,false', 'distance': '0,0', 'ifname': ',', 'nexthop-vrf': ','}```
Also I added same to ECMP with nexthop vrf - test_multiple_nexthops_with_vrf_static_route